scripts/machine-launch-script.sh

```diff
@@ -1,83 +1,260 @@
-sudo python3 -m pip install pyparsing==3.0.6
-sudo python3 -m pip install numpy==1.19.5
-sudo python3 -m pip install kiwisolver==1.3.1
-sudo python3 -m pip install matplotlib==3.3.4
-sudo python3 -m pip install pandas==1.1.5
-sudo python3 -m pip install awscli==1.22.21
-sudo python3 -m pip install pytest==6.2.5
-sudo python3 -m pip install moto==2.2.17
-# needed for the awstools cmdline parsing
-sudo python3 -m pip install pyyaml==5.4.1
-
-# setup argcomplete
-activate-global-python-argcomplete
-
-} 2>&1 | tee /home/centos/machine-launchstatus.log
-
-# get a regular prompt
-echo "PS1='\u@\H:\w\\$ '" >> /home/centos/.bashrc
-echo "machine launch script completed" >> /home/centos/machine-launchstatus
+
+    if [[ ! -r /etc/os-release ]]; then
+	echo "::ERROR:: $0 depends on /etc/os-release for distro-specific setup and it doesn't exist here"
+	exit 1
+    fi
+
+    if [[ "$(uname -io)" != "x86_64 GNU/Linux" ]]; then
+	echo "::ERROR:: $0 only supports 'x86_64 GNU/Linux' not '$(uname -io)'"
+	exit 1
+    fi
+
+    OS_FLAVOR=$(grep '^ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"')
+    OS_VERSION=$(grep '^VERSION_ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"')
+
+    echo "machine launch script started" > machine-launchstatus
+    chmod ugo+r machine-launchstatus
+
+    # platform-specific setup 
+    case "$OS_FLAVOR" in
+	ubuntu)
+	    # default /bin/sh to bash (not dash)
+	    sudo dpkg-reconfigure -p critical dash
+	    ;;
+	centos)
+	    ;;
+	*)
+	    echo "::WARNING:: Unknown OS flavor '$OS_FLAVOR'. Unable to do platform-specific setup."
+	    ;;
+    esac
+
+
+    # everything else is platform-agnostic and could easily be expanded to Windows and/or OSX
+
+    SUDO=""
+    prefix_parent=$(dirname "$CONDA_INSTALL_PREFIX")
+    if [[ ! -e "$prefix_parent" ]]; then
+	mkdir -p "$prefix_parent" || SUDO=sudo
+    elif [[ ! -w "$prefix_parent" ]]; then
+	SUDO=sudo
+    fi
+
+    if [[ -n "$SUDO" ]]; then
+	echo "::INFO:: using 'sudo' to install conda"
+	# ensure files are read-execute for everyone
+	umask 022
+    fi
+
+    # to enable use of sudo and avoid modifying 'secure_path' in /etc/sudoers, we specify the full path to conda
+    CONDA_EXE="${CONDA_INSTALL_PREFIX}/bin/conda"
+
+    if [[ -x "$CONDA_EXE" ]]; then
+	echo "::INFO:: '$CONDA_EXE' already exists, skipping conda install"
+    else
+	wget -O install_conda.sh "$CONDA_INSTALLER"  || curl -fsSLo install_conda.sh "$CONDA_INSTALLER"
+	echo "::INFO:: installing conda to '$CONDA_INSTALL_PREFIX'"
+	# -b for non-interactive install
+	$SUDO bash ./install_conda.sh -b -p "$CONDA_INSTALL_PREFIX"
+    fi
+
+    # see https://conda-forge.org/docs/user/tipsandtricks.html#multiple-channels
+    # for more information on strict channel_priority
+    $SUDO "$CONDA_EXE" config --system --set channel_priority strict
+    # By default, don't mess with people's PS1, I personally find it annoying
+    $SUDO "$CONDA_EXE" config --system --set changeps1 false
```

Comment:
```suggestion

        # see https://conda-forge.org/docs/user/tipsandtricks.html#multiple-channels
        # for more information on strict channel_priority
        $SUDO "$CONDA_EXE" config --system --set channel_priority strict
        # By default, don't mess with people's PS1, I personally find it annoying
        $SUDO "$CONDA_EXE" config --system --set changeps1 false
    fi
```

Hoist the system-wide conda-config stuff into only happening if we are doing the conda install